book/player/quickbrick-tv-player.md

```diff
@@ -0,0 +1,306 @@
+# Player Plugin for tvOS / AndroidTV / FireTV
+
+This document explains how to create a player plugin for the Quick-Brick framework.
+
+## Create your plugin project
+
+In order to have a working environment for your plugin, you will need several things:
+
+- A repository for your plugin code.
+- Native source code for the app you are working with, for that you can use the [downloaded development project](/getting-started/download-development-project.md).
+
+## Specific requirements
+
+Because the Quick-Brick app is built with react-native, and react-native is still implemented in objective-C on iOS, it is important that the main player view class and all the properties declared below are annotated with `@objc` and declared as public.
+
+## Anatomy of a QuickBrick player plugin
+
+The main export of your player plugin should simply be a view, extending `UIView` on **tvOS**, and `FrameLayout` on **AndroidTV/FireTV**.
+
+Your view must have the following properties below.
+
+Everytime the react-native code is setting or changing a property in the react-native component, it invokes the setter function of the relevant property. Depending on the value of that property, you can trigger whatever action is required on the native side (play / pause, mute / unmute, seek, etc...)
+
+Typically, the javascript front end will look something like this (there are more props and component methods, but this is just an example)
+
+```javascript
+// we get a reference to the underlying native component.
+// You don't have to create a bridge, it's built-in, and will instantiate
+// your player view by reflexion
+const NativeVideoComponent = requireNativeComponent("NativePlayerManager");
```

Comment:
You are referring to a native module, where is the native side of the `NativePlayerManager`?

Reply:
@snooky23 the `NativePlayerManager` is only in the RN side, it is the one that managed the bridges with the native player. The implementation on the native side is described below.
@f-roland I got it right?

Reply:
@lomudi yes. It is a native class, but it's part of the React Native boilerplate.
Its purpose is only to bridge the properties and callback between the js and the native code. 
That part of the RN bridge, `NativePlayerManager` needs to return a View from one of it's method. That's where we'll instantiate the player view from the plugin (by reflexion), so that all the pieces will be tied out together

I mainly provided the js example to show how the props on the js side map out to properties on the view class (of primitive types or RCTBubblingEventBlock for functions/callbacks). But if that line is confusing it can be ellipsed, as it is not necessary to understand